broker-daemon/utils/grpc-gateway.js

```diff
@@ -1,43 +1,73 @@
 /**
- * @author konsumer
- * @see {@link https://github.com/konsumer/grpc-dynamic-gateway}
+ * Sparkswap GRPC Gateway HTTP Proxy
+ *
+ * This loads a proto file and creates REST endpoints from it's google.api.http declarations.
+ *
+ * We 'forked' this library from the origin author konsumer which can be found at
+ * the link below.
  *
+ * @see {@link https://github.com/konsumer/grpc-dynamic-gateway}
  */
 
-const requiredGrpc = require('grpc')
+const grpc = require('grpc')
 const express = require('express')
 require('colors')
 const fs = require('fs')
 const schema = require('protocol-buffers-schema')
 
-const supportedMethods = ['get', 'put', 'post', 'delete', 'patch'] // supported HTTP methods
-const paramRegex = /{(\w+)}/g // regex to find gRPC params in url
+/**
+ * Supported HTTP methods for proto file
+ * @constant
+ * @type {Array}
+ * @default
+ */
+const supportedMethods = Object.freeze([
+  'get',
+  'put',
+  'post',
+  'delete',
+  'patch'
+])
+
+/**
+ * Regexp to find gRPC params in url
+ * @constant
+ * @type {RegExp}
+ * @default
+ */
+const paramRegex = /{(\w+)}/g
+
+/**
+ * @constant
+ * @type {String}
+ * @default
+ */
+const GRPC_API_OPTION_ID = '.google.api.http'
 
-const lowerFirstChar = str => str.charAt(0).toLowerCase() + str.slice(1)
+/**
+ * Takes a string (from grpc proto file) and formats to camelCase
+ *
+ * @param {String} str
+ * @returns {String} string formatted in camelCase
+ */
+function lowerFirstChar (str) {
+  return str.charAt(0).toLowerCase() + str.slice(1)
+}
 
 /**
  * generate middleware to proxy to gRPC defined by proto files
  * @param  {string[]} protoFiles Filenames of protobuf-file
  * @param  {string} grpcLocation HOST:PORT of gRPC server
  * @param  {ChannelCredentials}  gRPC credential context (default: grpc.credentials.createInsecure())
- * @param  {string} include      Path to find all includes
  * @return {Function}            Middleware
  */
-const middleware = (protoFiles, grpcLocation, credentials = requiredGrpc.credentials.createInsecure(), debug = true, include = '', grpc = requiredGrpc) => {
+const middleware = (protoFiles, grpcLocation, credentials = grpc.credentials.createInsecure(), debug = true) => {
   const router = express.Router()
   const clients = {}
-  if (include.endsWith('/')) {
-    include = include.substring(0, include.length - 1) // remove"/"
-  }
-  protoFiles = protoFiles.map(function (value, index, array) {
-    if (value.startsWith(include)) {
-      value = value.substring(include.length + 1)
-    }
-    return value
-  })
-  const protos = protoFiles.map(p => include ? grpc.load({ file: p, root: include }) : grpc.load(p))
```

Comment:
I removed the lines w/ `include` because our version of grpc.load does not take an object (it will fail).